### PR TITLE
feat: exposing methods for extensions

### PR DESCRIPTION
**Pull request by silver-ymz**

### CODE REVIEW

File: src/changelog.rs [hunk 1]
```diff
@@ -10,6 +10,8 @@ use crate::*;
 ///  - Fixed [`fs::ReadDir`]: Implement `PinnedDrop` to poll future stored internally,
 ///    otherwise it would drop the internal request ids too early, causing read task to fail
 ///    when they should not fail.
+/// ## Added
+///  - Exposing some methods in [`Sftp`] to check available extensions
```
Comment (NobodyXu) [inline src/changelog.rs:14]: Can you please list details of the added apis, e.g.

```suggestion
///  - Add new fn `Sftp::support_limits`
///  - Add new fn `Sftp::support_expand_path`
```

**Review by NobodyXu — CHANGES_REQUESTED** [review]

File: src/sftp.rs [hunk 2]
```diff
@@ -383,6 +383,54 @@ impl Sftp {
...
+
+    /// Check if the remote server supports the posix rename extension.
+    pub fn support_posix_rename(&self) -> bool {
+        self.handle
+            .get_auxiliary()
+            .extensions()
+            .contains(Extensions::POSIX_RENAME)
+    }
+
+    /// Check if the remote server supports the copy data extension.
```
Comment (NobodyXu) [inline src/sftp.rs:427]: Can we also add doc to corresponded function?

E.g. for this extension:

```suggestion
    /// Check if the remote server supports the copy data extension.
    /// If it returns true, then `file::File::copy_all_to` is supported.
```

And also update doc for `file::File::copy_all_to` to point to this for checking whether it is supported?

**Review by NobodyXu — CHANGES_REQUESTED** [review]

File: src/sftp.rs [hunk 3]
```diff
@@ -385,6 +385,8 @@ impl Sftp {
     }
 
     /// Check if the remote server supports the limits extension.
+    ///
+    /// If it returns true, then the client will follow the limits set by the server.
     pub fn support_limits(&self) -> bool {
```
Comment (NobodyXu) [inline src/sftp.rs:390]: We don't need to expose this since it is only used internally.

**Review by NobodyXu — CHANGES_REQUESTED** [review]

File: src/sftp.rs (hunk 3 above)
Comment (silver-ymz) [reply src/sftp.rs:390]: So should we remove `support_limits` api?

Comment (NobodyXu) [reply src/sftp.rs]: Yes

**Review by NobodyXu — APPROVED** [review]

**Pull request merged**